File: data-structure/stack/stack.py
from sys import maxsize

def createStack():
    stack = []
    return stack

def isEmpty(stack):
    return len(stack)

def push(stack, item):
    stack.append(item) # append 함수 이용

def pop(stack):
    if(isEmpty(stack)):
        return str(-maxsize -1) # stack underflow 안나게
    return stack.pop

def peek(stack):
    if (isEmpty(stack)):
        return str(-maxsize -1)
    return stack[len(stack) - 1]

stack = createStack()
push(stack, 10)
push(stack, 20)
push(stack, 30)

print(stack) # 최하단 원소부터 출력
print(stack[::-1]) # 최상단 원소부터 출력

File: data-structure/stack/stack.cpp
#include <iostream>
using namespace std;

#define MAX 1000 // Maximum size of Stack 정의

class Stack {
    int top;

    public:
        int a[MAX];
        Stack() {top = -1;} // initialization
        bool push(int x);
        int pop();
        int peek();
        bool isEmpty();
};

bool Stack::push(int x){
    if(top >= (MAX - 1)) {
        cout << "Stack overflow";
        return false;
    }
    else {
        a[++top] = x;
        return true;
    }
}

int Stack::pop(){
    if(top < 0){
        cout << "Stack underflow";
        return 0;
    }
    else {
        int x = a[top--];
        return x;
    }
}

int Stack::peek(){
    if(top < 0){
        cout << "빈 스택입니다이";
        return 0;
    }
    else {
        int x = a[top];
        return x;
    }
}

bool Stack::isEmpty () {
    return (top < 0);
}

int main () {
    class Stack s;
    s.push(10);
    s.push(20);
    s.push(30);
    cout << s.pop();

    // 스택 출력
    while (!s.isEmpty()){
        cout << " " << s.peek();
        s.pop();
    }
    return 0;
}
